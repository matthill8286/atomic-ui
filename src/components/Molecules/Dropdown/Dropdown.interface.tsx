import React from 'react'

export type DropDownListType = 'custom' | 'native'

export interface DropdownProps {
  autoComplete?: string
  initialIndex?: number
  label?: string
  listType?: DropDownListType
  onChange?: (index: number) => void
  options: DropdownOptionProps[]
}

export interface DropdownPropsEnhanced extends DropdownProps {
  showDropdown: boolean
  toggleDropdown: () => void
}

export interface ListProps {
  active: boolean
  isSearchable: boolean
}

export interface StyledDropdownProps {
  hasLabel: boolean
}

export interface StyledDropdownOptionProps {
  active: boolean
  isFocused: boolean
  onClick: React.MouseEventHandler<HTMLLIElement>
}

export interface DropdownOptionProps {
  id?: string
  isFocused?: boolean
  label: string
  url?: string
}

export interface SearchableDropdownProps {
  id?: string
  name?: string
  className?: string
  errorMessage?: string
  initializeOpen?: boolean
  inputValue: string | undefined
  label?: string
  onBlur?: React.FocusEventHandler<HTMLInputElement>
  onInputChange: React.ChangeEventHandler<HTMLInputElement>
  onSelectChange: (item: DropdownOptionProps) => void
  options: DropdownOptionProps[]
  selectedOption: string
  helpText?: string
  inputRef?: React.RefObject<HTMLInputElement>
}
