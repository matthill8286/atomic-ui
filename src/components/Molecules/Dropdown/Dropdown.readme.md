## Dropdown Component ##

Select components are used for collecting user provided information from a list of options.

#### When To Use ####

A dropdown menu for displaying choices - an elegant alternative to the native <select> element.

Utilizing Radio is recommended when there are fewer total options (less than 4 or 5).

#### Changes ####

Placeholder and label had been merged into label prop.
The label behave like a placeholder when the dropdown hasn't a selected value, then after the user select an element it moves into the top left corner.

The prop resetToInitialIndex had been removed, the component will check and set the initialIndex to the currentIndex when it is specified.
