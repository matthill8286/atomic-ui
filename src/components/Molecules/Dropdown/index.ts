export * from './Dropdown'
export * from './Dropdown.interface'
export * from './SearchableDropdown'
export * from './NativeDropdown'
