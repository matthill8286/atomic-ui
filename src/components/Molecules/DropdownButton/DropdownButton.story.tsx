import { action } from '@storybook/addon-actions'
import { storiesOf } from '@storybook/react'
import * as React from 'react'
import { StyleguideGlobe, IconUser } from '@excelwithbusiness/webmobile-svg-library'
import { DropdownButton } from './DropdownButton'

storiesOf('Design System/Molecules/Dropdown Button', module)
  .add('Default', () => (
    <DropdownButton isOpen={false} label="Dropdown Label" onClick={action('Button Clicked')} />
  ))
  .add('User Button', () => (
    <DropdownButton
      width={35}
      height={35}
      primaryIcon={<IconUser height={35} width={35} />}
      isOpen={false}
      onClick={action('Button Clicked')}
    />
  ))
  .add('User Button with picture', () => (
    <DropdownButton
      width={35}
      height={35}
      isOpen={false}
      onClick={action('Button Clicked')}
      pictureSrc="https://image.shutterstock.com/image-photo/cute-little-red-kitten-sleeps-600w-235089946.jpg"
    />
  ))
  .add('Icon Language Button open', () => (
    <DropdownButton
      secondaryIcon
      isOpen
      onClick={action('Button Clicked')}
      primaryIcon={<StyleguideGlobe />}
    />
  ))
  .add('Icon Language Button closed', () => (
    <DropdownButton
      secondaryIcon
      isOpen={false}
      onClick={action('Button Clicked')}
      primaryIcon={<StyleguideGlobe />}
    />
  ))
