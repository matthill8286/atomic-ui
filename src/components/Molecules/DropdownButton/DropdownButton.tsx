import React, { FunctionComponent } from 'react'
import { Card } from '@/components/Atoms/Card'
import { Icon } from '@/components/Atoms/Icon'
import { CopyText } from '@/components/Atoms/Typography'
import { css, styled } from '@/styles'
import { StyleguideArrow } from '@matthill8286/jsx-icon-library'
import { Button } from '@/components/Atoms/Button'
import { Picture } from '@/components/Atoms/Picture'
import { isNonEmptyString } from '@/components/Helper/String'

export interface DropdownButtonProps {
  label?: string
  onClick: React.MouseEventHandler
  additionalLabel?: string
  isOpen: boolean
  width?: string | number
  height?: string | number
  primaryIcon?: JSX.Element
  secondaryIcon?: boolean
  className?: string
  pictureSrc?: string
}

export const StyledContainer = styled.div`
  order: 3;
  position: relative;
`

export const StyledDropdown = styled(Card).attrs(() => ({
  elevation: 2,
  shape: 'rounded-small',
  noBorder: 'none',
  padding: '0',
}))<{ showDropdown: boolean; isMobile: boolean }>`
  position: ${({ isMobile }) => (isMobile ? `relative` : `absolute`)};
  left: -100px;
  top: ${({ theme, isMobile }) => (isMobile ? 0 : theme.spacing.base.xxxl)};
  width: 280px;
  margin: ${({ theme }) => theme.spacing.base.sm} 0 0 0;
  opacity: 0;
  visibility: hidden;
  z-index: 99;
  transform: translateY(-${({ theme }) => theme.spacing.base.sm});
  transition: ${({
    theme: {
      transition: { short, defaultEasing },
    },
  }) =>
    `visibility ${short} ${defaultEasing}, opacity ${short} ${defaultEasing}, transform ${short} ${defaultEasing}`};

  ${({ showDropdown }) =>
    showDropdown
      ? css`
          opacity: 1;
          visibility: visible;
          transform: translateY(0);
        `
      : ''}
`

const StyledButton = styled(Button)`
  padding: 0 ${({ theme }) => theme.spacing.base.xs};
  align-items: center;
  background-color: transparent;
  border-color: transparent;
  border-radius: ${({ theme }) => theme.dimension.borderRadius1};
  cursor: pointer;
  :active,
  :focus {
    outline: none;
  }

  :hover {
    /* box-shadow: ${({ theme }) => theme.dimension.elevationLevel1}; */
    box-shadow: none;
  }

  ${Icon} {
    padding: 0;
  }

  @media screen and (-ms-high-contrast: active), screen and (-ms-high-contrast: none) {
    /* IE10+ specific styles go here */
    width: 130px;
  }

  &::-moz-focus-inner {
    border: 0;
  }
`

const StyledContentGrid = styled.div`
  display: flex;
  flex-wrap: wrap;
`

const StyledLabel = styled(CopyText).attrs(() => ({
  tag: 'span',
  bold: true,
  color: 'grey5',
  fontSize: 'sm',
}))`
  justify-self: start;
  white-space: nowrap;
`

const StyledIcon = styled(Icon).attrs(() => ({
  color: 'grey5',
  alignContent: 'center',
}))`
  margin: 0;
  align-self: center;
`

const StyledAdditionalLabel = styled(CopyText).attrs(() => ({
  tag: 'span',
  color: 'grey5',
  lineHeight: 'sm',
}))`
  grid-column: 1 / 3;
  justify-self: start;
  overflow: hidden;
  white-space: nowrap;
  text-overflow: ellipsis;
  max-width: 9em;
  min-height: ${({ theme }) => theme.font.lineHeight.sm};
`

const StyledPictureElement = styled(Picture)`
  img {
    border-radius: 50%;
  }
`

export const DropdownButton: FunctionComponent<DropdownButtonProps> = ({
  onClick,
  additionalLabel,
  isOpen = false,
  label,
  height,
  width,
  secondaryIcon = false,
  primaryIcon,
  className,
  pictureSrc,
}) => (
  <StyledButton
    className={className}
    actionType="ghost"
    onClick={onClick as React.MouseEventHandler<HTMLElement>}>
    <StyledContentGrid>
      <StyledLabel>{label}</StyledLabel>
      {!isNonEmptyString(pictureSrc) && primaryIcon && (
        <StyledIcon
          height={(height as number) || (width as number)}
          width={width as number}
          padding="0">
          {primaryIcon}
        </StyledIcon>
      )}
      {isNonEmptyString(pictureSrc) && (
        <StyledPictureElement
          height={(height as string) || (width as string)}
          width={width as string}
          src={pictureSrc}
          alt="Picture"
        />
      )}
      {secondaryIcon && (
        <StyledIcon width={16} height={16} rotate={isOpen ? 270 : 90} animate>
          <StyleguideArrow />
        </StyledIcon>
      )}
      {additionalLabel && <StyledAdditionalLabel>{additionalLabel}</StyledAdditionalLabel>}
    </StyledContentGrid>
  </StyledButton>
)
