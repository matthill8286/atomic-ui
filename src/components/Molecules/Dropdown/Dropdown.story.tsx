import { action } from '@storybook/addon-actions'
import { number, text } from '@storybook/addon-knobs'
import { storiesOf } from '@storybook/react'
import React from 'react'
import { Dropdown } from './Dropdown'
import { DropdownProps } from './Dropdown.interface'
import readme from './Dropdown.readme.md'

const options = [
  { id: '0', label: 'BMW' },
  { id: '1', label: 'Audi' },
  { id: '2', label: 'Mercedes' },
  { id: '3', label: 'Porsche' },
  { id: '4', label: 'Very long text item that is long rly rly long long long' },
]

const commonKnobs = {
  onChange: action('on-change'),
  options,
}

const story = storiesOf('Design System/Molecules/ Dropdown', module)

story.add(
  'Initial',
  () => {
    const knobs: DropdownProps = {
      ...commonKnobs,
      label: text('label', 'type a label'),
    }

    return <Dropdown {...knobs} />
  },
  { info: readme }
)

story.add(
  'Selected',
  () => {
    const knobs: DropdownProps = {
      ...commonKnobs,
      label: text('Label', 'type a label'),
      initialIndex: number('initialIndex', 0),
    }

    return <Dropdown {...knobs} />
  },
  { info: readme }
)

story.add(
  'Native list',
  () => {
    const knobs: DropdownProps = {
      ...commonKnobs,
      label: text('Label', 'type a label'),
      initialIndex: number('initialIndex', 0),
      listType: 'native',
    }

    return <Dropdown {...knobs} />
  },
  { info: readme }
)
