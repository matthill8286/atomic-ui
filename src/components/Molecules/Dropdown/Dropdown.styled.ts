import { Icon } from '@/components/Atoms/Icon'
import { Typo } from '@/components/Atoms/Typography/Typo/Typo'
import { styled } from '@/styles/styled'
import { ListProps, StyledDropdownOptionProps, StyledDropdownProps } from './Dropdown.interface'

export const StyledContainer = styled.div`
  position: relative;
  cursor: pointer;

  & label {
    margin-top: ${({ theme }) => theme.spacing.base.xs};
    line-height: ${({ theme }) => theme.font.lineHeight.xs};
    position: absolute;
    transform-origin: top left;
    transition: color 200ms cubic-bezier(0, 0, 0.2, 1) 0ms, top 200ms cubic-bezier(0, 0, 0.2, 1) 0ms;
  }

  & [data-label-filled='true'] {
    top: 0;
    font-size: 12px;
    transform-origin: top left;
  }

  & [data-label-filled='false'] {
    top: 16px;
    transform-origin: top left;
  }
`

export const StyledDropdown = styled.div<StyledDropdownProps>`
  width: 100%;
  position: relative;
  display: inline-block;
  margin-top: ${({ theme, hasLabel }) => (hasLabel ? theme.font.lineHeight.sm : 0)};
  height: ${({ theme }) => theme.font.lineHeight.sm};
  line-height: ${({ theme }) => theme.font.lineHeight.sm};

  & span {
    position: absolute;
    left: 0;
  }

  & + hr {
    height: 2px;
    width: 100%;
    margin-top: 2px;
  }
`

export const StyledTypo = styled(Typo)`
  max-width: calc(100% - 16px);
`

export const StyledIcon = styled(Icon)`
  pointer-events: none;
  position: absolute;
  top: 0;
  right: 0;
`

export const StyledList = styled.ol<ListProps>`
  ${({ active, isSearchable, theme }) => `
    min-width: 100%;
    max-width: 100%;
    max-height: 320px;
    overflow-y: scroll;
    position: absolute;
    top: ${isSearchable ? `56px` : `32px`};
    z-index: 3;
    list-style: none;
    padding: ${theme.spacing.base.xs} 0;
    margin: 0;
    box-shadow: ${theme.dimension.elevationLevel2};
    background-color: white;
    border-radius: ${theme.dimension.borderRadius1};
    transform: scaleY(0);
    transform: ${active && 'scaleY(1)'};
    transform-origin: 0 0;
    transition: transform ${theme.transition.short} ease;
  `}
`

export const StyledDropdownOption = styled.li<StyledDropdownOptionProps>`
  ${({ isFocused, theme }) => `
    display: block;
    max-width: 100%;
    padding: ${theme.spacing.base.xs} ${theme.spacing.base.sm};
    background: ${theme.color.white};
    hyphens: auto;
    text-align: left;

    &:hover {
      cursor: pointer;
      background: ${theme.color.grey1};
    }

    ${
      isFocused
        ? `
        cursor: pointer;
        background: ${theme.color.grey1};
      `
        : ''
    }
  `}
`

export const StyledSearchableDropdownWrapper = styled.div`
  position: relative;
`

export const StyledFieldWrapper = styled.div<StyledDropdownProps>`
  display: inline-block;
  margin-top: ${({ hasLabel, theme }) => (hasLabel ? theme.spacing.base.md : 0)};
  position: relative;
  width: 100%;
`

export const StyledDropdownSelect = styled.select`
  appearance: none;
  background-color: transparent;
  border: none;
  box-shadow: none;
  cursor: pointer;
  display: block;
  font-size: ${({ theme }) => theme.font.size.sm};
  font-family: ${({ theme }) => theme.font.family.default};
  height: ${({ theme }) => theme.spacing.base.md};
  outline: none;
  overflow: hidden;
  text-overflow: ellipsis;
  white-space: nowrap;
  width: 100%;
  z-index: 2;
`
