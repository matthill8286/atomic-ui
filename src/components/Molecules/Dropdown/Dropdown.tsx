import React, { FC, useState } from 'react'
import { InputDivider as Divider } from '@/components/Atoms/Input/InputDivider'
import { CopyText } from '@/components/Atoms/Typography'
import { withDropdownState } from '@/components/Helper/withDropdownState'
import { DropdownPropsEnhanced } from './Dropdown.interface'
import {
  StyledContainer,
  StyledDropdown,
  StyledDropdownSelect,
  StyledFieldWrapper,
  StyledIcon,
  StyledList,
  StyledTypo,
} from './Dropdown.styled'
import { DropdownOption } from './DropdownOption'
import { IconArrow } from '@matthill8286/atomic-icon-library'

const DropdownWithoutState: FC<DropdownPropsEnhanced> = props => {
  const {
    autoComplete,
    initialIndex,
    label,
    listType = 'custom',
    onChange,
    options = [],
    showDropdown,
    toggleDropdown,
  } = props

  let setupCurrentIndex = -1
  const lastIndexOfOptions = options.length - 1

  if (initialIndex === 0 || !!initialIndex) {
    if (initialIndex >= 0 && initialIndex <= lastIndexOfOptions) {
      setupCurrentIndex = initialIndex
    }
  }

  const [currentIndex, setCurrentIndex] = useState<number>(setupCurrentIndex)
  const [isFilled, setFill] = useState<boolean>(() => currentIndex !== -1)

  const handleItemClick = (index: number) => () => {
    setCurrentIndex(index)
    setFill(true)
    if (onChange) onChange(index)
  }

  const handleSelectChange = (ev: React.ChangeEvent<HTMLSelectElement>): void => {
    const index = options.findIndex(({ id }) => id === ev.target.value)
    if (index !== -1) {
      handleItemClick(index)()
    }
  }

  const selectedItem = options.find((o, i) => i === currentIndex)

  return (
    <StyledContainer onClick={toggleDropdown}>
      {label && (
        <CopyText tag="label" color={isFilled ? 'grey5' : 'grey2'} data-label-filled={isFilled}>
          {label}
        </CopyText>
      )}
      {listType === 'custom' ? (
        <StyledDropdown hasLabel={!!label}>
          {currentIndex !== -1 && (
            <StyledTypo fontSize="sm" limitLines={1} lineHeight="sm" tag="span">
              {options[currentIndex].label}
            </StyledTypo>
          )}
          <StyledIcon width={16} height={16} rotate={showDropdown ? 270 : 90}>
            <IconArrow />
          </StyledIcon>

          <StyledList active={!!showDropdown} isSearchable={false}>
            {options.map((item, index) => (
              <DropdownOption
                key={index}
                isFocused={false}
                active={index === currentIndex}
                label={item.label}
                onClick={handleItemClick(index)}
              />
            ))}
          </StyledList>
        </StyledDropdown>
      ) : (
        <StyledFieldWrapper hasLabel={!!label}>
          <StyledDropdownSelect
            onChange={handleSelectChange}
            {...(selectedItem?.id ? { value: selectedItem.id } : {})}
            {...(autoComplete ? { autoComplete } : {})}>
            {options.map((item, i) => {
              return (
                <option key={item.id || `dropdown-${i}`} value={item.id}>
                  {item.label}
                </option>
              )
            })}
          </StyledDropdownSelect>
          <StyledIcon width={16} height={16} rotate={90}>
            <IconArrow />
          </StyledIcon>
        </StyledFieldWrapper>
      )}
      <Divider color="grey3" />
    </StyledContainer>
  )
}

export const Dropdown = withDropdownState(DropdownWithoutState, { displayName: 'Dropdown' })
