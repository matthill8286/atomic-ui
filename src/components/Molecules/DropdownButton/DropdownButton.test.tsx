import React from 'react'
import { mountWithTheme, renderWithTheme } from '@/testRenderer'
import { IconUser } from '@matthill8286/jsx-icon-library'
import { Picture } from '@/components/Atoms/Picture'
import { DropdownButton } from './DropdownButton'
import { Dropdown } from '../Dropdown'

const onClick = jest.fn()

describe('DropdownButton', () => {
  it('renders', () => {
    const wrapper = renderWithTheme(
      <DropdownButton width={35} height={35} isOpen={false} onClick={onClick} />
    )
    expect(wrapper).toMatchSnapshot()
  })
  it('should be clickable', () => {
    const wrapper = mountWithTheme(
      <DropdownButton width={35} height={35} isOpen={false} onClick={onClick} />
    )
    wrapper.simulate('click')
    const options = <Dropdown options={[{ label: 'one' }, { label: 'two' }]} />
    expect(options).toBeDefined()
  })
  it('renders isOpen prop as false', () => {
    const wrapper = mountWithTheme(
      <DropdownButton width={35} height={35} isOpen={false} onClick={onClick} />
    )
    expect(wrapper.find(DropdownButton).prop('isOpen')).toEqual(false)
  })
  it('render profile icon', () => {
    const wrapper = mountWithTheme(
      <DropdownButton
        width={35}
        height={35}
        isOpen={false}
        primaryIcon={<IconUser height={35} width={35} />}
        onClick={onClick}
      />
    )
    expect(wrapper.find(IconUser)).toHaveLength(1)
  })
  it('render profile picture', () => {
    const wrapper = mountWithTheme(
      <DropdownButton
        width={35}
        height={35}
        isOpen={false}
        pictureSrc="https://media.graphcms.com/xlJatgRxRASTguH6kE3A"
        onClick={onClick}
      />
    )
    expect(wrapper.find(Picture)).toHaveLength(1)
  })
})
