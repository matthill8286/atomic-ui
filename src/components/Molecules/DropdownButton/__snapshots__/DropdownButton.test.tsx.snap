// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DropdownButton renders 1`] = `
.c3 {
  color: #666666;
  font-size: 16px;
  line-height: 24px;
  font-family: "Oswald","Raleway",Arial,sans-serif;
  font-weight: 700;
  -webkit-letter-spacing: normal;
  -moz-letter-spacing: normal;
  -ms-letter-spacing: normal;
  letter-spacing: normal;
}

.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  border-radius: 4px;
  box-sizing: border-box;
  outline: none;
  cursor: pointer;
  -webkit-appearance: none;
  -moz-appearance: none;
  appearance: none;
  -webkit-appearance: none;
  padding: 12px 24px;
  position: relative;
  -webkit-text-decoration: none;
  text-decoration: none;
  font-family: "Oswald","Raleway",Arial,sans-serif;
  font-weight: normal;
  font-size: 16px;
  height: 48px;
  width: -webkit-fit-content;
  width: -moz-fit-content;
  width: fit-content;
  box-shadow: none;
  border: 2px solid #FFFFFF;
  background: #FFFFFF;
  color: #666666;
}

.c0 .Icon-sc-1ate1r7-0 {
  padding: 0 8px;
}

.c0 .Status-ekucpk-0 {
  position: absolute;
  top: -6px;
  right: -6px;
}

.c0:hover {
  box-shadow: 0 8px 10px 1px rgba(0,0,0,0.07),0 3px 14px 2px rgba(0,0,0,0.06),0 5px 5px -3px rgba(0,0,0,0.10);
}

.c0:hover {
  box-shadow: 0 8px 10px 1px rgba(0,0,0,0.07),0 3px 14px 2px rgba(0,0,0,0.06),0 5px 5px -3px rgba(0,0,0,0.10);
}

.c0 > span {
  color: inherit;
  display: block;
  text-align: center;
  width: 100%;
}

.c1 {
  padding: 0 8px;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  background-color: transparent;
  border-color: transparent;
  border-radius: 4px;
  cursor: pointer;
}

.c1:active,
.c1:focus {
  outline: none;
}

.c1:hover {
  box-shadow: none;
}

.c1 .Icon-sc-1ate1r7-0 {
  padding: 0;
}

.c1::-moz-focus-inner {
  border: 0;
}

.c2 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-wrap: wrap;
  -ms-flex-wrap: wrap;
  flex-wrap: wrap;
}

.c4 {
  justify-self: start;
  white-space: nowrap;
}

@media screen and (-ms-high-contrast:active),screen and (-ms-high-contrast:none) {
  .c1 {
    width: 130px;
  }
}

<button
  className="c0 c1"
  disabled={false}
  onClick={[Function]}
  size="md"
  type="button"
>
  <div
    className="c2"
  >
    <span
      className="c3 c4"
      color="grey5"
      fontSize="sm"
    />
  </div>
</button>
`;
